core/src/main/java/com/linecorp/armeria/client/Endpoint.java

```diff
@@ -433,6 +438,96 @@ public String authority() {
         return this.authority = authority;
     }
 
+    /**
+     * Converts this endpoint into a URI using the {@code scheme}.
+     *
+     * @param scheme the {@code scheme} for {@link URI}.
+     *
+     * @return the URI
+     */
+    public URI toUri(String scheme) {
+        requireNonNull(scheme, "scheme");
+
+        return toUri(Scheme.parse(scheme));
+    }
+
+    /**
+     * Converts this endpoint into a URI using the {@code scheme} and {@code path}.
+     *
+     * @param scheme the {@code scheme} for {@link URI}.
+     * @param path the {@code path} for {@link URI}.
+     *
+     * @return the URI
+     */
+    public URI toUri(String scheme, String path) {
+        requireNonNull(scheme, "scheme");
+        requireNonNull(path, "path");
+
+        return toUri(Scheme.parse(scheme), path);
+    }
+
+    /**
+     * Converts this endpoint into a URI using the {@link SessionProtocol} and {@link SerializationFormat}.
+     *
+     * @param sessionProtocol the {@link SessionProtocol} for {@link URI}.
+     * @param serializationFormat the {@link SerializationFormat} for {@link URI}
+     *
+     * @return the URI
+     */
+    public URI toUri(SessionProtocol sessionProtocol, SerializationFormat serializationFormat) {
+        requireNonNull(serializationFormat, "serializationFormat");
+        requireNonNull(sessionProtocol, "sessionProtocol");
+
+        return toUri(Scheme.of(serializationFormat, sessionProtocol));
+    }
+
+    /**
+     * Converts this endpoint into a URI using the {@link SessionProtocol}, {@link SerializationFormat}
+     * and {@code path}.
+     *
+     * @param sessionProtocol the {@link SessionProtocol} for {@link URI}.
+     * @param serializationFormat the {@link SerializationFormat} for {@link URI}
+     * @param path the {@code path} for {@link URI}.
+     *
+     * @return the URI
+     */
+    public URI toUri(SessionProtocol sessionProtocol, SerializationFormat serializationFormat, String path) {
```

Comment:
- How about removing `SerializationFormat` parameter and making this method to a shortcut of `toUri(Scheme.of(NONE, sessionProtocol), path)`?
- `path` could be `@Nullable`.